modules/light-clients/08-wasm/internal/types/store.go

```diff
@@ -0,0 +1,226 @@
+package types
+
+import (
+	"bytes"
+	"errors"
+	"io"
+
+	wasmvmtypes "github.com/CosmWasm/wasmvm/v2/types"
+
+	"cosmossdk.io/store/cachekv"
+	"cosmossdk.io/store/tracekv"
+	storetypes "cosmossdk.io/store/types"
+)
+
+var (
+	_ wasmvmtypes.KVStore = &StoreAdapter{}
+	_ storetypes.KVStore  = &MigrateClientWrappedStore{}
+
+	SubjectPrefix    = []byte("subject/")
+	SubstitutePrefix = []byte("substitute/")
+)
+
+// MigrateClientWrappedStore combines two KVStores into one.
+//
+// Both stores are used for reads, but only the subjectStore is used for writes. For all operations, the key
+// is checked to determine which types to use and must be prefixed with either "subject/" or "substitute/" accordingly.
+// If the key is not prefixed with either "subject/" or "substitute/", a default action is taken (e.g. no-op for Set/Delete).
+type MigrateClientWrappedStore struct {
+	subjectStore    storetypes.KVStore
+	substituteStore storetypes.KVStore
+}
+
+func NewMigrateClientWrappedStore(subjectStore, substituteStore storetypes.KVStore) MigrateClientWrappedStore {
```

Comment:
nit: basic godoc would be good here, sometimes linters cry about godocs on exported symbols